txnkv/transaction/txn.go

```diff
@@ -656,10 +861,11 @@ func (txn *KVTxn) LockKeys(ctx context.Context, lockCtx *tikv.LockCtx, keysInput
 				return txn.committer.extractKeyExistsErr(e)
 			}
 		}
+		keyStr := string(key)
```

Comment:
Would this introduce allocation in the recent golang version? Seems it still needs to https://github.com/golang/go/issues/25484?

Reply:
It's the simplest way, as it's impossible in golang to use slices as the builtin map's key. Here the purpose is to index the lockCtx.Value, which already exists before and uses string as the key too.